### PR TITLE
feat(angularls): add support for legacy projects

### PR DESCRIPTION
**Pull request by mjlbach**

### CODE REVIEW

**Comment by mjlbach** [comment]
@mactep sorry for the ping but can you test this?

**Comment by mactep** [comment]
On my way!

**Comment by mjlbach** [comment]
Great thanks :) I will merge before tagging the first release next tuesday if it works for you.

**Comment by mactep** [comment]
I'm setting up bare new projects with Angular 6 and Angular 13, which is the latest version.
<details>
  <summary>Here is my <code>init.vim</code> for reference</summary>

```vim
let data_dir = has('nvim') ? stdpath('data') . '/site' : '~/.vim'
if empty(glob(data_dir . '/autoload/plug.vim'))
  silent execute '!curl -fLo '.data_dir.'/autoload/plug.vim --create-dirs  https://raw.githubusercontent.com/junegunn/vim-plug/master/plug.vim'
  autocmd VimEnter * PlugInstall --sync | source $MYVIMRC
endif

call plug#begin(stdpath('data') . '/plugged')

Plug 'mjlbach/nvim-lspconfig', {'branch': 'feat/angular-legacy-support'}

call plug#end()

lua << EOF
require'lspconfig'.angularls.setup{}
EOF
```
</details>

Following the [documentation](https://github.com/mjlbach/nvim-lspconfig/blob/feat/angular-legacy-support/doc/server_configurations.md#angularls), I installed the language server globally and added the setup line to `init.vim`.

When opening both project I get `Client 1 quit with exit code 1 and signal 0`. Running `:LspInfo` I get:

```
 Config: angularls
 	filetypes:         typescript, html, typescriptreact, typescript.tsx
 	root directory:    /tmp/angularNewest
 	cmd:               ngserver --stdio
 	cmd is executable: true
 	autostart:         true
 	custom handlers:   
 
 Configured servers list: angularls
```

It looks like the `on_new_config` bit is not working properly.

<details>
  <summary>If we run <code>ngserver --stdio</code> inside both projects we get an error</summary>

```
/home/mactep/.local/lib/node_modules/@angular/language-server/index.js:219
      throw new Error(`Failed to resolve '${packageName}' with minimum version '${minVersion}' from ` + JSON.stringify(probeLocations, null, 2));
      ^

Error: Failed to resolve 'typescript/lib/tsserverlibrary' with minimum version '4.3' from []
    at resolveWithMinVersion (/home/mactep/.local/lib/node_modules/@angular/language-server/index.js:219:13)
    at resolveTsServer (/home/mactep/.local/lib/node_modules/@angular/language-server/index.js:228:14)
    at requireOverride (/home/mactep/.local/lib/node_modules/@angular/language-server/index.js:297:57)
    at dist/server/logger.js (/home/mactep/.local/lib/node_modules/@angular/language-server/index.js:414:27)
    at __require (/home/mactep/.local/lib/node_modules/@angular/language-server/index.js:313:44)
    at Object.<anonymous> (/home/mactep/.local/lib/node_modules/@angular/language-server/index.js:4557:16)
    at Module._compile (internal/modules/cjs/loader.js:1085:14)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)
    at Module.load (internal/modules/cjs/loader.js:950:32)
    at Function.Module._load (internal/modules/cjs/loader.js:790:12)
```
</details>

If we manually set the custom cmd to `ngserver --stdio --tsProbeLocations ./node_modules/ --ngProbeLocations ./node_modules/` it'll probe the current project for `typescript` and the `@angular/language-service` package. The first one should be already be installed and the second can be installed through `npm install --save-dev @angular/language-service`. That configuration works in the project with the newest version, but not on the older one. Running `:LspInfo` inside the older project shows that it didn't add the `--viewEngine` bit to the cmd. Adding that wouldn't work either since the language server needs a Typescript version that's greater than the version on the older project. So, we should update the documentation to provide the support for automatically adding the `--viewEngine` bit and and add more info about the server dependencies.

In conclusion, I think there was an error with the `--viewEngine` not being added and the way we install the language server. For the latter I think we could install `@angular/language-server`, `typescript` and `@angular/language-service` globally and add that directory as the default one. I'm doing a research about that to see if it should be better to install everything inside the project according to the required versions or install globally.

**Comment by mactep** [comment]
Update: as per the official [documentation](https://github.com/angular/vscode-ng-language-service#versioning), it looks like we can add to our documentation that all those 3 packages needs to installed and we probe a default global directory.

**Comment by mjlbach** [comment]
Can you give me a reference old/new angular project I can test? I'm not sure why this version wouldn't work

> So, we should update the documentation to provide the support for automatically adding the --viewEngine bit and and add more info about the server dependencies.

Sorry, I'm having trouble following, that should be what this PR does?

**Comment by mactep** [comment]
> Can you give me a reference old/new angular project I can test? I'm not sure why this version wouldn't work
To set up the testing projects I just did `npm i @angular/cli@6`, then `npx ng new angular6` then I installed the newest cli version `npm i @angular/cli` and `npx ng new angularNewest`.

> Sorry, I'm having trouble following, that should be what this PR does?
Kind of. I just don't know if this solution would manage custom setup of the language server. So there should be a snippet on the documentation I guess.
